util/i2csvg/README.md

```diff
@@ -0,0 +1,159 @@
+# i2csvg -- Generate svg pictures of i2c commands from text description
+
+i2csvg is a Python tool to read i2c transactions from a text file and
+generate `svg` pictures of them.
+
+Two source representations are supported.
+The first is a more readable format and describes transactions in text.
+The second is a hex-dump of values written to the I2C fifo.
+In either format a line starting with a `#` is taken as a comment and a line staring with a `T` will be used as a title in the output.
+
+To assist in parsing debug output the tool can be given a prefix.
+Only lines with the prefix will be processed (after removal of the prefix).
+
+### Textual Source Format
+
+The textual description includes the flags and value written to the I2C host interface fifo to make the transaction.
+The flags and data may be separated by whitespace.
+The flags are letters representing each of the 5 control bits written to the fifo:
+
+|Flag | Use |
+|-----|-----|
+|S | Send a start signal before the data |
+|P | Send a stop signal after the data |
+|R | Read transaction (data value indicates how many bytes to read) |
+|C | With R, allows read to continue by ACK of the last byte read |
+|N | NACK OK, do not raise an error on a NACK |
+
+The value may be an integer (binary, octal, decimal and hex supported) or one of the special values that change how the output is drawn:
+
+|Value | Use |
+|------|-----|
+|A0 | Address with direction bit set to 0 (write) |
+|A1 | Address with direction bit set to 1 (read) |
+|A2 | Address in Quick command with direction bit used for data |
```

Comment:
@mdhayter: I'm unfamiliar with the "Quick command" format.  Is this an SMBus term?  As far as I can tell it's not in the I2C base spec.

Reply:
Yes, this is an SMBus thing. 6.5.1 in http://smbus.org/specs/SMBus_3_0_20141220.pdf

I worked through all the commands in that chapter and all seem possible (the only quirk being a split in the ones that need to read N part way though)